src/data/files.rs

```diff
@@ -2,6 +2,59 @@ use crate::prelude::*;
 use nu_errors::ShellError;
 use nu_protocol::{TaggedDictBuilder, UntaggedValue, Value};
 
+pub(crate) fn empty_dir_entry_dict(
```

Comment:
Another possibility to this approach might be something like a function that creates the dir_entry based on the values you pass in. You could keep the settings as a single struct that goes in as well.

This way you could reuse the same function between the `empty` version and the one filled with data for the dir_entry.

Reply:
I'll toss around an idea:

What if `insert_untagged` could take an `Option<V>` where `V: Into<Value>`? `None` would be a `Nothing`, otherwise we insert `V.into()`. Could that make it easy to have a single function that does both?